### DISCUSSION TITLE
Attaching to an element in the DOM

### DISCUSSION THREAD

**Discussion by jackmcdade** [floating-ui/floating-ui#2212] [Q&A]
I'm using Popper 2 in a Vue.js context, and in [the docs](https://popper.js.org/docs/v2/performance/#attaching-elements-to-the-dom) it mentions you can attach Popper.js instances to the DOM, but never explains how.

> To ensure the best possible performance, it's [recommended](https://web.archive.org/web/20210827084020/https://atfzl.com/don-t-attach-tooltips-to-document-body) to not attach elements directly to the `<body />` element, but instead create a `<div />` element right within the `<body />` and use it as container of your poppers.

Furthermore, in the Floating UI docs it mentions you can append to the body while portaling, but again — no explaination of how.
https://floating-ui.com/docs/misc#clipping

I need to be able to move Popper instances to a consistent place in the DOM, but still relative to their triggering element. Is this possible? For the life of me I can't find an example.

**Comment by atomiks** [comment]
I think this is the Vue API: https://vuejs.org/guide/built-ins/teleport.html

**Reply by jackmcdade** [reply]
Ah, we're using Vue 2, which doesn't have that. We're using a Portal library, but that still doesn't explain how to actually use it and get the positioning correct. It's always rendered in the wrong location — usually way off to the right.

I've tried to address that with `offset`, but until the popper instance is rendered we don't know the width, so we can't calculate it dynamically.

**Reply by atomiks** [reply]
I don't know what code you have, but try this:

- Make sure you have `<!DOCTYPE html>` in your page
- Make sure the portal wrapper from the library doesn't have any conflicting styles that could cause the wrong position
- Make sure you're on `@popperjs/core@2.11.6`

**Reply by jackmcdade** [reply]
Check check check on all 3.

**Reply by atomiks** [reply]
I'm afraid I can't help further unless you can provide a minimal reproduction with CodeSandbox/Stackblitz demonstrating the problem. By doing so you might find what's wrong also.

You can also try Floating UI's Vue wrapper to see if it works: https://floating-ui.com/docs/vue - it contains several bug fixes that Popper doesn't have; though Popper is used in production in many websites and big component libs and problems relating to portals haven't been reported.

**Comment by jackmcdade** [comment]
Willing to pay for a solution, I'm so frustrated and stuck that I'll do almost anything to solve this.
